src/main/scala/co/teapot/tempest/graph/MemMappedDynamicUnidirectionalGraph.scala

```diff
@@ -59,27 +59,35 @@ class MemMappedDynamicUnidirectionalGraph(allocator: MemoryMappedAllocator,
   //   3) a (12 * nodeCapacity)-byte array of node data
   //        (4-byte degree and an 8-byte neighborPointer per node)
 
-  def dataPointer = data.getLong(dataPointerPointer)
-  private def setDataPointer(p: Long): Unit = data.putLong(dataPointerPointer, p)
+  def dataPointer: Pointer = data.getPointer(dataPointerPointer)
+  private def setDataPointer(p: Pointer): Unit = data.putPointer(dataPointerPointer, p)
 
   // We store maxNodeId as a long to support ids between 2**31 and 2**32.
-  def maxNodeId: Int = maxNodeIdLong.toInt
+  def maxNodeId: Int = {
+    val result = maxNodeIdLong
+    require(result < Int.MaxValue)
+    result.toInt
+  }
   def maxNodeIdLong: Long =
     data.getLong(dataPointer + MaxNodeIdOffset)
   private def setMaxNodeId(id: Long): Unit = {
     data.putLong(dataPointer + MaxNodeIdOffset, id, allocator.syncAllWrites)
   }
 
-  private def nodeCapacity: Long = data.getLong(dataPointer + NodeCapacityOffset)
+  private def nodeCapacity: Int = {
+    val result = data.getLong(dataPointer + NodeCapacityOffset)
+    require(result < Int.MaxValue)
+    result.toInt
+  }
   private def setNodeCapacity(c: Long): Unit = data.putLong(dataPointer + NodeCapacityOffset, c, allocator.syncAllWrites)
 
-  private def nodeArrayPointer: Long = dataPointer + NodeArrayOffset
+  private def nodeArrayPointer: Pointer = dataPointer + NodeArrayOffset
 
   /** Returns a pointer to a node's (degree, neighborPointer) pair.
     * */
-  private def nodePointer(id: Int): Long = {
+  private def nodePointer(id: Int): Pointer = {
     require(Integer.toUnsignedLong(id) < nodeCapacity, s"id ${Integer.toUnsignedLong(id)} >= nodeCapacity $nodeCapacity")
-    nodeArrayPointer + BytesPerNode * Integer.toUnsignedLong(id)
+    nodeArrayPointer + Offset.fromBlocks(id, BytesPerNode)
```

Comment:
`Integer.toUnsignedLong` was to support having between 2^31 and 2^32 nodes.  It was pretty hacky though, so I'm fine with this PR removing `Integer.toUnsignedLong`.